packages/server/modules/shared/index.js

```diff
@@ -1,46 +1,58 @@
 'use strict'
-const Redis = require( 'ioredis' )
-const debug = require( 'debug' )( 'speckle:middleware' )
-const appRoot = require( 'app-root-path' )
-const knex = require( `${appRoot}/db/knex` )
-const { ForbiddenError, ApolloError } = require( 'apollo-server-express' )
-const { RedisPubSub } = require( 'graphql-redis-subscriptions' )
-const { validateToken } = require( `${appRoot}/modules/core/services/tokens` )
-
-
-let pubsub = new RedisPubSub( {
-  publisher: new Redis( process.env.REDIS_URL ),
-  subscriber: new Redis( process.env.REDIS_URL ),
-} )
+const Redis = require('ioredis')
+const debug = require('debug')('speckle:middleware')
+const appRoot = require('app-root-path')
+const knex = require(`@/db/knex`)
+const { ForbiddenError, ApolloError } = require('apollo-server-express')
+const { RedisPubSub } = require('graphql-redis-subscriptions')
+const { buildRequestLoaders } = require('@/modules/core/loaders')
+const { validateToken } = require(`${appRoot}/modules/core/services/tokens`)
+
+let pubsub = new RedisPubSub({
+  publisher: new Redis(process.env.REDIS_URL),
+  subscriber: new Redis(process.env.REDIS_URL)
+})
+
+async function buildContext({ req, connection }) {
```

Comment:
1. better name for a function that builds the GQL context
2. added request-scoped dataloaders onto the ctx object, so that they can be accessed in any resolver